### DISCUSSION TITLE
articles/git-paging

### DISCUSSION THREAD

**Discussion by hutusi** [hutusi/hutusi.github.com#133] [Comments]
https://hutusi.com/articles/git-paging 

 通过阅读源码来学习开源项目是最直接也最有效的方法。而想要了解一个开源项目，最好是从第一个 commit 开始看起，特别是第一个最小可用版本发布前的 commits, 通过阅读对每一次 commit 提交源码，能够最直接的理解作者的设计思路和开发过程中的思考。

**Comment by wojiaohanshuai** [comment]
你好，文章中说 git-first 和 git-next 这俩个脚本转为可执行文件，linux环境下我搜到了解决方案（有 shc 和 gzexe），请问知道如何在windows环境下操作吗？

**Comment by hutusi** [comment]
> @wojiaohanshuai
> 你好，文章中说 git-first 和 git-next 这俩个脚本转为可执行文件，linux环境下我搜到了解决方案（有 shc 和 gzexe），请问知道如何在windows环境下操作吗？

谢谢你的提醒，文章说的转为可执行文件有点歧义，实际上是加可执行权限即可。 `chmod +x`， gist 我已经更新，clone 下来便已有可执行权限。

Windows 增加了 tips： Windows 的 Git 一般安装在 C:\Program Files\Git 下，bin 目录非根目录下的 bin, 而是在 usr/bin 下，直接将 git-first 等文件拷贝至该目录即可。（注意：该目录原来有 git-flow 等文件。）

有问题的话欢迎交流 :)

**Comment by Kavinjsir** [comment]
我用的是 `tig` ，可以比较方便地浏览git历史记录， `tig reverse` 可以从头开始。

**Comment by hutusi** [comment]
@Kavinjsir 谢谢你的方法，我也试试`tig`这个工具看看~~
